noxfile.py

```diff
@@ -234,7 +249,22 @@ def prerelease_deps(session):
         "pytest",
         "pytest-cov",
     )
-    session.install("-e", ".[all]")
+
+    with open(
+        CURRENT_DIRECTORY / "testing" / "constraints-3.6.txt", encoding="utf-8"
+    ) as constraints_file:
+        constraints_text = constraints_file.read()
+    constraints_text = re.sub(r"==\S+", "\n", constraints_text)
+    deps = [
+        line.strip()
+        for line in constraints_text.split("\n")
+        if line.strip() and line[0] != "#"
```

Comment:
What if we somehow end up with a comment line that starts with whitespace? Unlikely, indeed, but it would feel clumsy if we accidentally break the pipeline with redundant whitespace in a text file. :laughing: 

BTW, should we instead match the entire lines containing version pins and directly extract the dependecy name with grouping in the regex pattern itself? Feels more straightforward, to me at lest.

Reply:
I like it! I made one modification, which was to add `^` to only match from the start of a line. Otherwise, it picked up "foo" from the comments.